### DISCUSSION TITLE
Stream HiGHs solver output in pyomo

### DISCUSSION THREAD

**Discussion by ckaldemeyer** [ERGO-Code/HiGHS#1835] [General]
Thanks for starting and creating this wonderful project!

I am using `pyomo` to model certain problems and would like to use the `tee=True` flag to stream the solver output.

Unfortunately, there seems to be an issue which is related to the interaction between HiGHs and the global interpreter lock as described here:
- https://github.com/Pyomo/pyomo/issues/3084
- https://github.com/Pyomo/pyomo/issues/3031

Would this be easy to fix?

Best wishes and thanks in advance,
Cord

**Comment by jajhall** [comment]
I've heard about this, and the logging callback in HiGHS
 https://ergo-code.github.io/HiGHS/dev/callbacks/

gives much more options for controlling output but, it's really up to Pyomo folk to do something about it. 

Naturally we're happy to field questions.
